lib/public-stats.js

```diff
@@ -274,3 +275,77 @@ const updateRetrievalTimings = async (pgClient, committees) => {
     JSON.stringify(rows)
   ])
 }
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealClients
+ */
+const updateDailyClientStats = async (pgClient, committees, findDealClients) => {
+  const clientRetrievalStats = new Map()
+  /** @type {Map<string, Array<Measurement>>} */
+  const clientToMeasurementsMap = new Map()
+  for (const c of committees) {
+    const { minerId, payload_cid: payloadCid } = c.retrievalTask
+
+    const clients = findDealClients(minerId, payloadCid)
+    if (!clients || !clients.length) {
+      console.warn(`Invalid retrieval task (${minerId}, ${payloadCid}): no deal clients found. Excluding the task from daily per-client stats.`)
+      Sentry.captureException(new Error('Invalid retrieval task: no deal clients found.'), {
+        extra: {
+          minerId,
+          payload_cid: payloadCid
+        }
+      })
+      continue
+    }
+    clients.forEach(clientId => {
+      if (clientToMeasurementsMap.has(clientId)) {
+        clientToMeasurementsMap.get(clientId).push(...c.measurements)
+      } else {
+        clientToMeasurementsMap.set(clientId, Array.from(c.measurements))
```

Comment:
```suggestion
        clientToMeasurementsMap.set(clientId, c.measurements)
```

`Committee#measurements` already is an array, see https://github.com/CheckerNetwork/spark-evaluate/blob/ba0ffe197774f7d19a298ecbfa4097109adf29c8/lib/committee.js#L20

Reply:
I am getting an error that says the object is not extensible if I do not wrap it in an Array.